core/arch/arm/kernel/generic_boot.c

```diff
@@ -166,6 +170,30 @@ static void init_vfp_sec(void)
 }
 #endif
 
+#ifdef CFG_INIT_CNTFRQ
+static void main_init_cntfrq(void)
+{
+	assert(cntfrq == 0);
+	assert(get_core_pos() == 0);
+	cntfrq = read_cntfrq();
```

Comment:
This solves the problem of syncing CNTFRQ during boot. However, it doesn't initialize CNTFRQ, but it depends on CNTFRQ being initialized. Is this supposed to be done in earlier boot stages or am I missing something?

Reply:
Yes, CNTFRQ is supposed to be initialized on the primary core by a previous boot stage. Perhaps I should rename the config option to `CFG_SECONDARY_INIT_CNTFRQ` ?

Reply:
Add a comment about what's expected from the previous boot stage would be nice.